### DISCUSSION TITLE
I want to use runtime-telemetry-java17.

### DISCUSSION THREAD

**Discussion by shinozakisatoshi11** [open-telemetry/opentelemetry-java-instrumentation#12831] [Q&A]
I have enabled otel.instrumentation.runtime-telemetry-java17.enabled=true. However, the otel_scope_name in the collected metrics indicates "java8". Is this expected behavior? For example, with metrics like [jvm_memory_used_bytes].

**Comment by laurit** [comment]
> Is this expected behavior?

Yes it is. I'd suggest you study the source code of the `runtime-telemetry-java17` to get a better understanding of what id does. By default it enables the metrics from `runtime-telemetry-java8` that are mostly gathered through jmx. There is an option to use jfr instead where the `runtime-telemetry-java17` will produce different metrics.

**Comment by shinozakisatoshi11** [comment]
Thanks.
 I was referring to this area, is that correct?　[https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5e28dfd8c71d0d8ff9fe9451b2828a3dac3c14c7/instrumentation/runtime-telemetry/runtime-telemetry-java17/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java17/RuntimeMetricsBuilder.java](url)
Also, I understood that the java8 one is JMX and the java17 one is the guy who gets it from jFR, is that correct?

**Reply by laurit** [reply]
> Also, I understood that the java8 one is JMX and the java17 one is the guy who gets it from jFR, is that correct?

To reiterate, by default `runtime-telemetry-java17` use the code from `runtime-telemetry-java8` that uses jmx. Jfr usage can be enabled with a flag.
